authentication/src/test/java/io/camunda/authentication/filters/AdminUserCheckFilterTest.java

```diff
@@ -0,0 +1,114 @@
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.authentication.filters;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.camunda.search.entities.RoleMemberEntity;
+import io.camunda.search.query.SearchQueryResult;
+import io.camunda.security.configuration.SecurityConfiguration;
+import io.camunda.service.RoleServices;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+
+class AdminUserCheckFilterTest {
+
+  @Test
+  void shouldRedirectIfNoAdminUserExistsOrIsConfigured() throws ServletException, IOException {
+    // given
+    final var securityConfig = new SecurityConfiguration();
+    final var roleServices = mock(RoleServices.class);
+    when(roleServices.searchMembers(any())).thenReturn(SearchQueryResult.empty());
+    final var request = mock(HttpServletRequest.class);
+    when(request.getRequestURI()).thenReturn("/identity/users");
+    when(request.getContextPath()).thenReturn("localhost:8080");
+    final var response = mock(HttpServletResponse.class);
+    final var filterChain = mock(FilterChain.class);
+    final AdminUserCheckFilter adminUserCheckFilter =
+        new AdminUserCheckFilter(securityConfig, roleServices);
+
+    // when
+    adminUserCheckFilter.doFilterInternal(request, response, filterChain);
+
+    // then
+    verify(response).sendRedirect("localhost:8080/identity/setup");
+  }
+
+  @Test
+  void shouldNotRedirectIfAdminUserIsConfigured() throws ServletException, IOException {
+    // given
+    final var securityConfig = new SecurityConfiguration();
+    securityConfig
+        .getInitialization()
+        .getDefaultRoles()
+        .put("admin", Map.of("users", Set.of("adminUser")));
+    final var roleServices = mock(RoleServices.class);
+    final var request = mock(HttpServletRequest.class);
+    when(request.getRequestURI()).thenReturn("/identity/users");
+    final var response = mock(HttpServletResponse.class);
+    final var filterChain = mock(FilterChain.class);
+    final AdminUserCheckFilter adminUserCheckFilter =
+        new AdminUserCheckFilter(securityConfig, roleServices);
+
+    // when
+    adminUserCheckFilter.doFilterInternal(request, response, filterChain);
+
+    // then
+    verify(filterChain).doFilter(request, response);
+  }
+
+  @Test
+  void shouldNotRedirectIfAdminUserExists() throws ServletException, IOException {
+    // given
+    final var securityConfig = new SecurityConfiguration();
```

Comment:
🔧 moving all the mock creations into a before might help with readability of the test case